core/embed/rust/src/ui/model_tt/component/keyboard/pin.rs

```diff
@@ -335,7 +337,7 @@ impl PinDots {
     }
 
     fn is_full(&self) -> bool {
-        self.digits.len() == self.digits.capacity()
+        self.digits.len() >= MAX_LENGTH
```

Comment:
What if `MAX_LENGTH`, defined at the beginning of the file is greater than the maximum length of `ShortString`? Should we check somewhere to ensure it isn't?

Reply:
https://github.com/trezor/trezor-firmware/pull/3951/commits/e63cb30d53b566cbd2ddaf097bb34d6232bd5368